tests/models/test_taskinstance.py

```diff
@@ -1714,10 +1714,19 @@ def test_outlet_datasets(self, create_task_instance, clear_datasets):
         ti.refresh_from_db()
         assert ti.state == TaskInstanceState.SUCCESS
 
+        # check that no other dataset events recorded
+        event = (
+            session.query(DatasetEvent)
+            .join(DatasetEvent.dataset)
+            .filter(DatasetEvent.source_task_instance == ti)
+            .one()
+        )
+        assert event
+        assert event.dataset
+
         # check that one queue record created for each dag that depends on dataset 1
-        assert session.query(DatasetDagRunQueue.target_dag_id).filter(
```

Comment:
This change is because SQLA was warning us about a (likely) unintended cross-product effect!